### PR TITLE
Rename database classes to make some sense

### LINKED ISSUE #8020: Rename storage layer objects to be more sensible
This is a recurring conversation so wanted to document some of it.

Some places this was talked about:

* [May 13, 2020](https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$KybugO1LibfFZZyuQyhFu0-pl-hLX2wIzwy3XFfaIaM?via=matrix.org&via=vector.modular.im&via=privacytools.io)
* [Jul 30, 2020](https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$b2LA02VVC9S_ylz1Hpk58I-rKLmr1UTxHVlsrSd3T4g?via=matrix.org&via=vector.modular.im&via=privacytools.io)
* [Aug 3, 2020](https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$s1IAkpxVFJ6knYLunJHEyPZSJhB-KxkgmSbH8JMZPq0?via=matrix.org&via=vector.modular.im&via=privacytools.io)

The current proposal is:

* Rename `synapse.storage.data_stores` to `synapse.storage.databases`
* Rename `Database` to `DatabasePool`

We would still have `DataStore` classes that make up a `database` (e.g. `database.state` or `database.main`).

### PR DESCRIPTION
**Pull request by erikjohnston**
Apologies to everyone with open PRs that will be conflicted

Fixes #8020

### CODE REVIEW

**Comment by clokep** [comment]
Was this done via some automated tooling? Going through every file seems like it won't be feasible here.

**Comment by erikjohnston** [comment]
> Was this done via some automated tooling? Going through every file seems like it won't be feasible here.

The renaming of classes and modules used `rope`, the renaming of `store.db` was done manually.

**Comment by clokep** [comment]
There's a few more references to `data_stores`, I'm not sure if they're worth cleaning up or not:
* `synapse/config/database.py` (we wouldn't update the config value, but maybe the variable names?)
* `synapse/storage/databases/__init__.py` seems to use the above.
* `synapse/storage/prepare_database.py` seems to have a few variables named `data_stores`

I did some other searching and didn't find other references to these. So I think this is OK.

File: synapse/storage/prepare_database.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def _upgrade_existing_database(
         # First we find the directories to search in
         delta_dir = os.path.join(dir_path, "schema", "delta", str(v))
         directories = [delta_dir]
-        for data_store in data_stores:
+        for data_store in databases:
             directories.append(
                 os.path.join(
                     dir_path, "databases", data_store, "schema", "delta", str(v)
```
Comment (clokep) [inline synapse/storage/prepare_database.py]:
```suggestion
        for database in databases:
            directories.append(
                os.path.join(
                    dir_path, "databases", database, "schema", "delta", str(v)
```

**Pull request merged**